httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/HttpByteArrayCacheEntrySerializer.java

```diff
@@ -0,0 +1,395 @@
+        String lastKey = null;
+        for (final Header header : headers) {
+            if (header.getName().equals(SC_HEADER_NAME_VARIANT_MAP_KEY)) {
+                lastKey = header.getValue();
+                response.removeHeader(header);
+            } else if (header.getName().equals(SC_HEADER_NAME_VARIANT_MAP_VALUE)) {
+                if (lastKey == null) {
+                    throw new ResourceIOException("Found mismatched variant map key/value headers");
+                }
+                variantMap.put(lastKey, header.getValue());
+                lastKey = null;
+                response.removeHeader(header);
+            }
+        }
+
+        if (lastKey != null) {
+            throw new ResourceIOException("Found mismatched variant map key/value headers");
+        }
+
+        return variantMap;
+    }
+
+    /**
+     * Copy bytes from the given source buffer and input stream to the given output stream until end-of-file is reached.
+     *
+     * @param srcBuf Buffered input source
+     * @param src Unbuffered input source
+     * @param dest Output destination
+     * @throws IOException if an I/O error occurs
+     */
+    private static void copyBytes(final SessionInputBuffer srcBuf, final InputStream src, final OutputStream dest) throws IOException {
+        final byte[] buf = new byte[BUFFER_SIZE];
+        int lastBytesRead;
+        while ((lastBytesRead = srcBuf.read(buf, src)) != -1) {
+            dest.write(buf, 0, lastBytesRead);
+        }
+    }
+
+    /**
+     * Writer for SimpleHttpResponse.
+     *
+     * Copied from DefaultHttpResponseWriter, but wrapping a SimpleHttpResponse instead of a ClassicHttpResponse
+     */
+    // Seems like the DefaultHttpResponseWriter should be able to do this, but it doesn't seem to be able to
+    private class SimpleHttpResponseWriter extends AbstractMessageWriter<SimpleHttpResponse> {
+
+        public SimpleHttpResponseWriter() {
+            super(BasicLineFormatter.INSTANCE);
+        }
+
+        @Override
+        protected void writeHeadLine(
+                final SimpleHttpResponse message, final CharArrayBuffer lineBuf) {
+            final ProtocolVersion transportVersion = message.getVersion();
+            BasicLineFormatter.INSTANCE.formatStatusLine(lineBuf, new StatusLine(
+                    transportVersion != null ? transportVersion : HttpVersion.HTTP_1_1,
+                    message.getCode(),
+                    message.getReasonPhrase()));
+        }
+    }
+
+    /**
+     * Cache validity policy that always returns an age of 0.
+     *
+     * This prevents the Age header from being written to the cache (it does not make sense to cache it),
+     * and is the only thing the policy is used for in this case.
+     */
+    private class NoAgeCacheValidityPolicy extends CacheValidityPolicy {
+        @Override
+        public long getCurrentAgeSecs(final HttpCacheEntry entry, final Date now) {
```

Comment:
Please note that will soon move to `TimeValue`.

Reply:
OK, seems like it should be an easy change, but not one I can make quite yet.